### PR TITLE
feat(replay): Add Mobile Replay

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## Only Reviewed Content from other PRs in this branch

This PR contains all the changes needed for RN Mobile Replay. 

Future alphas and betas should be also released from this branch.

#skip-changelog

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -54,5 +54,5 @@ android {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    api 'io.sentry:sentry-android:7.11.0'
+    api 'io.sentry:sentry-android:7.12.0-alpha.4'
```
Comment (lucas-zimerman) [inline android/build.gradle:57]: Q: do you need to use this alpha release?

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "@sentry/react-native",
   "homepage": "https://github.com/getsentry/sentry-react-native",
   "repository": "https://github.com/getsentry/sentry-react-native",
-  "version": "5.25.0",
+  "version": "5.25.0-alpha.2",
```
Comment (lucas-zimerman) [inline package.json:5]: Q: do we need to use this alpha version?

File: samples/expo/app.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "slug": "sentry-react-native-expo-sample",
     "jsEngine": "hermes",
     "scheme": "sentry-expo-sample",
-    "version": "5.25.0",
+    "version": "5.25.0-alpha.2",
```
Comment (lucas-zimerman) [inline samples/expo/app.json:7]: Q: Do we need to use this alpha version

File: samples/expo/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sentry-react-native-expo-sample",
-  "version": "5.25.0",
+  "version": "5.25.0-alpha.2",
```
Comment (lucas-zimerman) [inline samples/expo/package.json:3]: Q: do we need to use this alpha version?

File: samples/react-native/android/app/build.gradle [hunk 5]
```diff
@@ -134,8 +134,8 @@ android {
         applicationId "io.sentry.reactnative.sample"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 19
-        versionName "5.25.0"
+        versionCode 15
+        versionName "5.25.0-alpha.2"
```
Comment (lucas-zimerman) [inline samples/react-native/android/app/build.gradle:138]: Q: Do we need to use this alpha version?

File: samples/react-native/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sentry-react-native-sample",
-  "version": "5.25.0",
+  "version": "5.25.0-alpha.2",
```
Comment (lucas-zimerman) [inline samples/react-native/package.json:3]: Q: Do we need to use this alpha version?

File: src/js/version.ts [hunk 7]
```diff
@@ -1,3 +1,3 @@
 export const SDK_PACKAGE_NAME = 'npm:@sentry/react-native';
 export const SDK_NAME = 'sentry.javascript.react-native';
-export const SDK_VERSION = '5.25.0';
+export const SDK_VERSION = '5.25.0-alpha.2';
```
Comment (lucas-zimerman) [inline src/js/version.ts:3]: Q: Do we need to use this alpha version?

**Review by lucas-zimerman — APPROVED** [review]
Minor questions in regard versioning, but other than that it is not blocking if we change it or not so I am approving it :D

**Comment by krystofwoldrich** [comment]
The `5.25.0-alpha.2` is the latest version of Sentry's RN SDK with Session Replay, that's why it's in the branch, craft will overwrite it with a new version when we release it.

The Android Session Replay (Beta) will be released on Friday in a stable release of the Android SDK and we will release RN with that.

File: android/build.gradle (hunk 1 above)
Comment (krystofwoldrich) [reply android/build.gradle:57]: We will wait for stable release. Should be this Friday.

**Comment by lucas-zimerman** [comment]
Added a blocked label to remind that we should wait for non beta releases from the native SDKs to be released before merging.

File: android/build.gradle (hunk 1 above)
Comment (bruno-garcia) [reply android/build.gradle]: stable versions are out

**Pull request merged**